src/colorizer/utils/data_utils.ts

```diff
@@ -27,3 +28,45 @@ export function getColorMap(colorRampData: Map<string, ColorRampData>, key: stri
   }
   return colorRamp && reversed ? colorRamp.reverse() : colorRamp;
 }
+
+/**
+ * Validates the thresholds against the dataset. If the threshold's feature is present but the wrong type, it is updated.
+ * @param dataset The dataset to validate thresholds against.
+ * @param thresholds An array of `FeatureThresholds` to validate.
+ * @returns a new array of thresholds, with any categorical thresholds converted to numeric thresholds if the feature is numeric
+ * and vice-versa.
+ */
+export function validateThresholds(dataset: Dataset, thresholds: FeatureThreshold[]): FeatureThreshold[] {
+  // Validate feature data for each threshold. If the threshold is the wrong type, update it.
+  const newThresholds: FeatureThreshold[] = [];
+
+  for (const threshold of thresholds) {
+    const featureData = dataset.tryGetFeatureData(threshold.featureName);
+    const isInDataset = featureData && featureData.units === threshold.units;
+
+    if (isInDataset && featureData.type === FeatureType.CATEGORICAL && isThresholdNumeric(threshold)) {
+      // Threshold is not categorical but the feature is.
+      // Convert the threshold to categorical.
```

Comment:
what is the situation when these conditions can happen?  maybe give one example of why we need to do this in the jsdoc comment above?  I assume it's basically when switching datasets and trying to keep same filter settings?

Reply:
It's most likely to happen if users are opening links from older versions of Timelapse Colorizer (where categorical features are treated like numeric thresholds) or the case you mentioned, where one dataset labels a feature as categorical that another does not. I'll add an additional comment here.